legal-api/src/legal_api/reports/business_document.py

```diff
@@ -423,52 +421,51 @@ def _format_state_filing(self, filing: Filing) -> dict:
+        if filings:
+            amalgamation_application = filings[0]
             business['business']['amalgamatedEntity'] = True
-            amalgamation_json = amalgamation_application[0].filing_json.get('filing', {})\
-                .get('amalgamationApplication', {})
-            # if it's future effective
-            if self._epoch_filing_date and amalgamation_application[0].effective_date < self._epoch_filing_date:
+            if self._epoch_filing_date and amalgamation_application.effective_date < self._epoch_filing_date:
+                # imported from COLIN
                 amalgamated_businesses_info = {
                     'legalName': 'Not Available',
                     'identifier': 'Not Available'
                 }
                 amalgamated_businesses.append(amalgamated_businesses_info)
             else:
-                amalgamating_businesses = amalgamation_json.get('amalgamatingBusinesses', {})
-                while amalgamating_businesses:
-                    if len(amalgamating_businesses[0].get('foreignJurisdiction', {})) == 0:
-                        identifier = amalgamating_businesses[0].get('identifier', {})
+                amalgamation = self._business.amalgamation.one_or_none()
+                amalgamating_businesses = amalgamation.amalgamating_businesses.all()
+                for amalgamating_business in amalgamating_businesses:
+                    if ting_business := Business.find_by_internal_id(amalgamating_business.business_id):
+                        identifier = ting_business.identifier
+                        business_legal_name = ting_business.legal_name
                     else:
-                        identifier = amalgamating_businesses[0].get('corpNumber', {})
-                    amalgamating_business = Business.find_by_identifier(identifier)
-                    business_legal_name = amalgamating_business.legal_name
+                        identifier = amalgamating_business.foreign_corp_num or 'Not Available'
+                        business_legal_name = amalgamating_business.foreign_name or 'Not Available'
+
                     amalgamated_businesses_info = {
                         'legalName': business_legal_name,
                         'identifier': identifier
                     }
                     amalgamated_businesses.append(amalgamated_businesses_info)
-                    amalgamating_businesses.remove(amalgamating_businesses[0])
         business['amalgamatedEntities'] = amalgamated_businesses
 
     def _set_amalgamating_details(self, business: dict):
         amalgamating_info = business.get('business', {}).get('amalgamatedInto')
-        amalgamated_entity = business.get('business', {}).get('amalgamatedEntity')
         if amalgamating_info:
             business['business']['isAmalgamating'] = True
-            if amalgamated_entity:
-                business['business']['amalgamatedEntity'] = False
```

Comment:
`amalgamatedEntity` is used in `businessDetails.html`. using `amalgamatedEntities` instead